### DISCUSSION TITLE
Cross segment template properties in transient prompt

### DISCUSSION THREAD

**Discussion by dcnieho** [JanDeDobbeleer/oh-my-posh#3299] [Troubleshoot] [answered]
A pet peeve i have is that the time shown in a prompt is the time that the prompt became available, not when it was submitted. That makes for a not so ideal log. So i though, i could use transient prompts to rewrite part of the prompt upon submission and update the time to the time at which the prompt was submitted. But i'd like to keep other things intact. Since the transient prompt docs tell me `All template functionality is available, even reusing cross segment template properties from the previous primary prompt run` and since the docs of cross segment template properties give the example `{{ .Segments.Hello.Output }}`, I thought it would be easy to copy the parts of the prompt i want to remain in my log, and update some other parts with live information. (I am assuming .Output includes all the powerlines and everything by the way, if not, how to replicate that?)

However, I do not see the `.Output` entry. When i have the following segment:
```json
{
  "type": "executiontime",
  "background": "darkGray",
  "foreground": "lightWhite",
  "style": "diamond",
  "properties": {
    "style": "roundrock",
    "threshold": 0
  },
  "template": " \uf608 {{ .FormattedMs }} ",
  "trailing_diamond": "\ue0b4"
}
```
and I print whats in the `.Segments` map for Executiontime, i get: `map[FormattedMs:258ms Ms:258]`. When i try to print `{{  .Segments.Executiontime.Output  }}` I indeed get nothing. What am I missing?

This is on version 12.26.1, the latest version in the Windows store.

**Comment by JanDeDobbeleer** [comment, marked as answer]
@dcnieho you need `{{ .Segments.Executiontime.FormattedMs }}` or `{{ .Segments.Executiontime.Ms }}`. It says `Output` because in that example, the [command](https://ohmyposh.dev/docs/segments/command) segment has that [property](https://ohmyposh.dev/docs/segments/command#properties-1).

**Reply by dcnieho** [reply]
Yes, that works, but gives me just the text. Is there a way to reprint a whole segment as it would have been rendered?

As you have seen in some recent other issues i filed, i have a rather complicated prompt (and loving it!). I am hoping to use the transient prompt function to leave behind a simplified trace of it for history purposes, and as written above, update the time to the time the command was submitted. So copying some parts from the previous primary prompt run and formatting some other parts anew would be perfect. Is that all possible?

**Reply by JanDeDobbeleer** [reply]
@dcnieho no, you can't. What you can do for this one is recreate the same segment using a [text](https://ohmyposh.dev/docs/segments/text) segment and cross segment properties. That way nothing is recalculated.

**Reply by dcnieho** [reply]
@JanDeDobbeleer I'm not sure i understand your suggestion. I cannot add segments in my transient_prompt, right? This doesn't work it prints `pwsh>`:
```json
"transient_prompt": {
  "segments": [
    {
      "type": "text",
      "foreground": "white",
      "style": "plain",
      "template": "\u2570\u2500"
    }
  ]
},
```

**Reply by JanDeDobbeleer** [reply]
@dcnieho ah yes, transient prompt use-case is limited (due to obvious reasons, it should be smaller, not the same). So no you can't re re-reference rendered segments, that will be a world of pain to create too as printing happens sequentially.

**Reply by dcnieho** [reply]
hmm, no worries. Guess i'm trying to hack something together that isn't meant to be. I'll use a much simplified transient prompt, as what its meant for (e.g. easy copying of previous commands). Thanks!
